In [3]:
# import data
import pandas as pd

dfjeans = pd.read_excel("../data/jeans_data.xlsx")

In [4]:
dfjeans

,week,sales,price,special_event,se_indicator
0,1,230,114.75,Presidents Day,1
1,2,71,135.00,NaN,0
2,3,70,135.00,NaN,0
3,4,65,135.00,NaN,0
4,5,60,135.00,NaN,0
5,6,195,114.75,Easter,1
6,7,76,135.00,NaN,0
7,8,51,135.00,NaN,0
8,9,79,135.00,NaN,0
9,10,146,114.75,NaN,0


In [5]:
dfjeans.describe()

,week,sales,price,se_indicator
count,26.000000,26.000000,26.000000,26.000000
mean,13.500000,92.115385,128.769231,0.153846
std,7.648529,65.149875,12.063151,0.367946
min,1.000000,46.000000,101.250000,0.000000
25%,7.250000,51.000000,135.000000,0.000000
50%,13.500000,62.500000,135.000000,0.000000
75%,19.750000,78.250000,135.000000,0.000000
max,26.000000,253.000000,135.000000,1.000000


# Model 1 Regression - Price ($) and SE_indicator

In [44]:
# Regression using Price and Special Event indicator as predictors
import statsmodels.api as sm

# Add constant to feature matrix which includes price and se_indicator
X_model1 = sm.add_constant(dfjeans[["price","se_indicator"]].astype("float"))

# Fit the OLS model
olsjeans1 = sm.OLS(dfjeans.sales, X_model1).fit()

In [45]:
# Print summary
olsjeans1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sales   R-squared:                       0.963
Model:                            OLS   Adj. R-squared:                  0.959
Method:                 Least Squares   F-statistic:                     295.4
Date:                Wed, 25 Mar 2026   Prob (F-statistic):           3.96e-17
Time:                        09:49:26   Log-Likelihood:                -102.28
No. Observations:                  26   AIC:                             210.6
Df Residuals:                      23   BIC:                             214.3
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const          509.0259     43.675     11.655      0.000     418.677     599.375
price           -3.3287      0.329    -10.123      0.000      -4.009      -2.648
se_indicator    76.2267     10.781      7.071      0.000      53.925      98.528
==============================================================================
Omnibus:                        1.399   Durbin-Watson:                   1.189
Prob(Omnibus):                  0.497   Jarque-Bera (JB):                1.214
Skew:                           0.367   Prob(JB):                        0.545
Kurtosis:                       2.237   Cond. No.                     2.23e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 2.23e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

## Analysis of Model 1 Regression (Question a)

### Coefficients

Price returned with a coefficient of -3.31 while se_indicator had a coefficient of 76.23. We can interpret this as saying for each 1 unit increase of price ($1) we expect a decrease of about 3.31 units of sales, which lines up with common sense that as price increases, people's willingness to buy decreases (depending on the product).

On the other hand, the 76.23 coefficient on se_indicator shows that when there is a special event (se_indicator = 1) then we should expect sales to increase by 76, which shows a strong effect on the final sales number when relevant.

### P Values 
Both independent variables returned with a P-value of 0.000 (rounded to 3 digits), which indicates statistical significance. This would allow us to reject the null hypothesis that these two independent variables are not at least signficantly correleated with sales, since they are under the 0.05 threshold for 95% confidence.

### Adjusted R^2 score

For this regression, we have an adjusted R^2 score of .959, which can be interpreted as all of our predictors account for 95.9% of the variation observed in the dependent variable sales. This is quite a high score and gives at least a strong signal that this model includes features that are going to be useful for predicting sales effectively. Additionally, since it is the adjusted score, even with the penalty for including more terms (not that 2 terms is a large number by any means) it still scores highly, indicating a strong model for prediction.

## Predicting with Model 1

We can write the regression as:

Sales = 509.0259 + -3.3287 * Price + 76.2267 * se_indicator

Since our price is a $25 dollar discount, $135 - 25 = 110.

In [47]:
predicted_sales = olsjeans1.predict([1,110,1])

print(f'The predicted sales for the promotion is ${round(predicted_sales[0],2)}')

The predicted sales for the promotion is $219.09


# Model 2 Regression - Price as a discount and SE_indicator

In [48]:
# Create dummy variable for price as discount
dfjeans['discount'] = (135 - dfjeans.price) / 135

# Add constant to feature matrix which includes price and se_indicator
X_model2 = sm.add_constant(dfjeans[['discount','se_indicator']].astype("float"))

# Fit the OLS model
olsjeans2 = sm.OLS(dfjeans.sales, X_model2).fit()

In [49]:
olsjeans2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sales   R-squared:                       0.963
Model:                            OLS   Adj. R-squared:                  0.959
Method:                 Least Squares   F-statistic:                     295.4
Date:                Wed, 25 Mar 2026   Prob (F-statistic):           3.96e-17
Time:                        09:51:03   Log-Likelihood:                -102.28
No. Observations:                  26   AIC:                             210.6
Df Residuals:                      23   BIC:                             214.3
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const           59.6477      2.917     20.447      0.000      53.613      65.682
discount       449.3782     44.392     10.123      0.000     357.546     541.211
se_indicator    76.2267     10.781      7.071      0.000      53.925      98.528
==============================================================================
Omnibus:                        1.399   Durbin-Watson:                   1.189
Prob(Omnibus):                  0.497   Jarque-Bera (JB):                1.214
Skew:                           0.367   Prob(JB):                        0.545
Kurtosis:                       2.237   Cond. No.                         17.8
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

## Analysis of Model 2 Regression (Question b)

### Coefficients

In model 2, discount had a coefficient of 449.3782 while se_indicator had 76.2267. Similar to the first regression, these tell us how much change we can expect in sales when each variable moves by 1. In this case, neither of these independent variables actually move by 1 but have a maximum value of 1 and either equal 0 (se_indicator) or can range values between 0 and 1 (discount).

se_indicator is the same as model 1, which makes sense since by using the price as a discount, we are functionally just rearranging the way the values are written but not changing them in any way so the se_indicator impact is untouched.

### P Values 
All independent variables returned with a P-value of 0.000 (rounded to 3 digits), which indicates statistical significance. This would allow us to reject the null hypothesis that these two independent variables are not at least signficantly correleated with sales, since they are under the 0.05 threshold for 95% confidence.

This result is the same as model 1, so both models have independent variables that are all statistically significant for sales, indicating no clear advantage in that regard of one model over the other.

### Adjusted R^2 score

For regression 2, we have an adjusted R^2 score of 0.959, which can be interpreted as all of our predictors account for 95.7% of the variation observed in the dependent variable sales. This is quite a high score and gives at least a strong signal that this model includes features that are going to be useful for predicting sales effectively.

This is just slightly lower than the r^2 term in model 1 and while it is not much worse, it at least gives us a way to potentially compare and rank our models over the other. Since we are not doing any sort of train/test splitting and predicting in this case, if r^2 is the only metric we have to use to compare the two models, there would be a slight preference for model 1, since it has better predictive power for the variance in sales over model 2 according to the slightly higher r^2 score.

## Predicting with Model 2

We can write the regression as:

Sales = 59.6477 + 449.3782 * Discount + 76.2267 * se_indicator

Since our price is a $25 dollar discount, $135 - 25 = $110. 135 - 110 / 135 = 18.52%

135 - 110 / 135

In [53]:
predicted_sales = olsjeans2.predict([1,.1852,1])
print(f'The predicted sales for the promotion is ${round(predicted_sales[0],2)}')

The predicted sales for the promotion is $219.1
